app/helpers/suffix_tree_helper.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module SuffixTreeHelper
+  class SuffixTreeSingleton < SuffixTree
+    include Singleton
+    def initialize
+      super(AppConfig['suffix_tree']['str_path'], AppConfig['suffix_tree']['tag_path'],
+            AppConfig['suffix_tree']['child_path'], AppConfig['suffix_tree']['node_path'])
+      @functional = true
+      @reason = ''
+    end
+
+    def mark_functional
+      @functional = true
+    end
+
+    def mark_broken(reason)
+      @functional = false
+      @reason = reason
+    end
+
+    def functional?
+      @functional
+    end
+
+    def broken_reason
+      @reason
+    end
```

Comment:
I'd recommend just setting `attr_accessor :broken_reason` and then calling that accessor from `#mark_broken` and `#initialize` (although you don't actually need to initialize it, if you don't it'll just be `nil` which shouldn't cause any problems that I can see)